model.py

```diff
@@ -63,8 +63,20 @@ def train(validate=True, epochs=20):
     print("y_train Shape: ", y_train.shape)
     print("y_test Shape: ", y_test.shape)
 
-    img_rows, img_cols = 224, 224
-    vgg = tf.keras.applications.vgg16.VGG16(weights='imagenet', include_top=False, input_shape=(img_rows, img_cols, 3))
+    input_channels = 4 if use_nir else 3
+    input_shape = (224, 224, input_channels)
+
+    weights = 'imagenet' if input_channels == 3 else None
+
+    vgg = tf.keras.applications.vgg16.VGG16(
+        weights=weights,
+        include_top=False,
+        input_shape=input_shape
+    )
+    
+    '''
+    since VGG16 is pre-trained w/ 3-channel RGB images, this if-else ensure it runs on a 4-channel system
+    '''
```

Comment:
```suggestion
    
    # since VGG16 is pre-trained w/ 3-channel RGB images, this if-else ensure it runs on a 4-channel system
    
```
pep-8 style comments use # not ''' '''